_assets/js/laws-regs-parser.js

```diff
@@ -158,3 +155,61 @@ function shareLink(e, divType) {
         shareIcon.style.display = 'block';
     }, 2000);
 }
+
+function addCopyEventListeners(btn, divType) {
+    btn.addEventListener('click', (e) => { copyText(e, divType) });
+    btn.addEventListener('mouseover', (e) => { highlightText(e, divType) });
+    btn.addEventListener('mouseleave', (e) => { unHighlightText(e, divType) });
+}
+
+function addPrintEventListeners(btn, divType) {
+    btn.addEventListener('click', (e) => { printText(e, divType) });
+    btn.addEventListener('mouseover', (e) => { highlightText(e, divType) });
+    btn.addEventListener('mouseleave', (e) => { unHighlightText(e, divType) });
+}
+
+function search() {
+    const searchBox = document.getElementById('searchbox');
+    const searchQuery = searchBox.value.toLowerCase();
+    if (!searchQuery.length){
+        clearSearch();
+        return;
+    };
+    const sections = document.querySelectorAll('.section')
+    for (let i = 0; i < sections.length; i++) {
+        removeHighlights(sections[i]);
+        if(sections[i].innerText.toLowerCase()
+            .includes(searchQuery)) {
+            sections[i].classList.remove("hidden");
+            sections[i].classList.add("searched");
+            highlightTerm(searchQuery, sections[i]);
+        } else {
+            sections[i].classList.add("hidden");
+            sections[i].classList.remove("searched");
+        }
+    }
+}
+
+function highlightTerm(text, section) {
+    const innerHTML = section.innerHTML.replaceAll(new RegExp('(' + text + ')',"ig"), `<span class='search-term'>$1</span>`);
+    section.innerHTML = innerHTML;
+    const shareBtn = section.querySelector('.Share-btn');
```

Comment:
nit: unless existing code standardizes on uppercase class names, I generally coerce them to lowercase ([see this SO answer](https://stackoverflow.com/a/2345300))